packages/mui-system/src/DefaultPropsProvider/DefaultPropsProvider.tsx

```diff
@@ -0,0 +1,61 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import resolveProps from '@mui/utils/resolveProps';
+
+const PropsContext = React.createContext<Record<string, any> | undefined>(undefined);
```

Comment:
We are missing the codebase guideline for React context here:

```suggestion
const PropsContext = React.createContext<Record<string, any> | undefined>(undefined);

if (process.env.NODE_ENV !== 'production') {
  PropsContext.displayName = 'DefaultProps';
```

Reply:
Also, the name of the context looks wrong

```suggestion
const DefaultPropsContext = React.createContext<Record<string, any> | undefined>(undefined);
```